### PR TITLE
generalize SDK client initialization method for model providers

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description

- prevent init service provider's SDK clients in raw code, use generalized static method `get_client` in each model provider instead
- minimize the impact from future changes for client initialization

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @takatost

**Comment by bowenliang123** [comment]
Hi, after offline discussion, I have put the method as a mixin without extending ModelProvider class directly. Please have a check. cc @takatost

File: api/core/model_runtime/model_providers/__base/model_client_provider.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from typing import Any
+
+
+class ModelClientProvider:
+    @staticmethod
+    def get_service_client(credentials: dict = None, **kwargs: Any) -> Any:
```
Comment (takatost) [inline api/core/model_runtime/model_providers/__base/model_client_provider.py]: credentials parameter should be declared as optional

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/__base/model_client_provider.py (hunk 1 above)
Comment (bowenliang123) [reply api/core/model_runtime/model_providers/__base/model_client_provider.py]: OK, changed to `Optional[dict]`. And this PR has been rebased onto the latest commit of the master branch as well.

**Comment by bowenliang123** [comment]
Closing this PR as offline discussed.

**Pull request closed without merging**